apps/onboarding-functions/src/main/java/it/pagopa/selfcare/onboarding/functions/OnboardingFunctions.java

```diff
@@ -128,7 +140,12 @@ public void onboardingsOrchestrator(
                 default -> throw new IllegalArgumentException("Workflow options not found!");
             }
 
-            Optional<OnboardingStatus> optNextStatus = workflowExecutor.execute(ctx, onboarding);
+            Product product = productService.getProductIsValid(onboarding.getProductId());
+            OnboardingWorkflow onboardingWorkflow = new OnboardingWorkflow();
+            onboardingWorkflow.setOnboarding(onboarding);
+            onboardingWorkflow.setContractTemplatePath(workflowExecutor.getContractTemplatePath(product, onboarding));
+            onboardingWorkflow.setEmailRegistrationPath(workflowExecutor.getEmailRegistrationPath(mailTemplatePathConfig));
```

Comment:
it might be better to set this data when persist onboarding on onboarding-ms...this method is dedicated to orchestrate the workflow rather than to prepare data for business logic....in this way at each orchestration (REQUEST->PENDING, PENDING->COMPLETE or REQUEST->TOBEVALIDATE and TOBEVALIDATE->PENDING) you retrive product and set values each time